components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIProviderImpl.java

```diff
@@ -977,8 +992,13 @@ public API addAPI(API api) throws APIManagementException {
         }
 
         try {
-            PublisherAPI addedAPI = apiPersistenceInstance.addAPI(new Organization(tenantDomain),
-                    APIMapper.INSTANCE.toPublisherApi(api));
+            Organization org;
+            if (api.getOrganizationId() != null) {
+                org = new Organization(api.getOrganizationId());
+            } else {
+                org = new Organization(tenantDomain);
+            }
```

Comment:
How about setting an OrganizationDTO itself into API object.
The behaviour would be:
Choreo Org case:

OrganizationDTO.getOrgID = <choreo_org>
OrganizationDTO.getTenantDomain = carbon.super

On-premise

OrganizationDTO.getOrgID = <tenantDomain of API>
OrganizationDTO.getTenantDomain = <tenantDomain of API>

Then we can just get API.getOrganizationDTO().getOrgID() and assign to this.

Reply:
We need the **OrganizationDTO** object when we require both the tenantDomain as well as the **orgId**. 
In the cases where setting an **orgId** into API object, we don't need that extra attribute of tenantDomain.

Is there any special reason to pass an **OrganizationDTO** object instead of **orgId** ?